File: map/Interpreter2.0/model/PrintStatement.cs
using System;
using DataStructure;
namespace models
{
    public class PrintStatement : IntStatement
    {
        Expression exp;

        public PrintStatement(Expression _exp){
            this.exp = _exp;
        }


        override public String ToString() {
            return "print(" + exp.ToString() + ")";
        }

        public ProgState execute(ProgState state){
            MyIntList<int> dataout = state.getOut();
            try {
                dataout.add(exp.eval(state.getSymTable()));
            }
            catch(MyException.MyException e) {
                throw e;
            }
            return null;
        }
        
    }
}

File: map/InterpreterFX/src/model/PrintStatement.java
package model;

import DataStructure.IntHeap;
import DataStructure.MyIntList;
import ExceptionHandling.MyException;

public class PrintStatement implements IntStatement {
    Expresion exp;

    public PrintStatement(Expresion _exp){
        this.exp = _exp;
    }


    @Override
    public String toString() {
        return "print(" + exp.toString() + ")";
    }

    @Override
    public ProgState execute(ProgState state) throws MyException{
        MyIntList<Integer> out = state.getOut();
        IntHeap heap = state.getHeap();
        try {
            out.add(exp.eval(state.getSymTable().top(), heap));
        }
        catch(MyException e) {
            throw e;
        }
        return null;
    }
}
